carma_cloud_client/src/carma_cloud_client_node.cpp

```diff
@@ -497,79 +497,87 @@ namespace carma_cloud_client
   {
     j2735_v2x_msgs::msg::TrafficControlDetail tcm_detail;
 
-    auto child_closed = tree.get_child_optional( "detail.closed" );
-    if( child_closed )
+    auto detail_tree = tree.get_child("detail");
+
+
+    auto closed_tree = detail_tree.get_child_optional( "closed" );
+    if (closed_tree)
     {
       tcm_detail.choice = j2735_v2x_msgs::msg::TrafficControlDetail::CLOSED_CHOICE;
-      std::string closed_val = tree.get<std::string>("detail.closed");
-      if (closed_val == "open")
+
+      auto open = closed_tree.get().get_child_optional("open");
```

Comment:
Looks like we should be getting these from the detail_tree instead of closed. Can you verify if "open" is an element within closed? Or its own optional element